motto/agents/openrouter_agent.py

```diff
@@ -0,0 +1,73 @@
+from .metta_agent import Agent
+import os
+import requests
+import json
+
+key = os.environ.get('OPENROUTER_API_KEY')
+
+
+class Function:
+    def __init__(self, name, arguments):
+        self.name = name
+        self.arguments = arguments
+
+
+class ToolCalls:
+    def __init__(self, tool_id, call_type, function):
+        self.id = tool_id
+        self.type = call_type
+        self.function = function
+
+
+class MessageClass:
+    def __init__(self, role, content, tool_calls=None):
+        self.role = role
+        self.content = content
+        self.tool_calls = tool_calls
+
+
+class OpenRouterAgent(Agent):
+    '''
+    OpenRouter is a unified interface for LLMs, this agent use OpenRouter API to get responses from LLM
+    '''
+
+    def __init__(self, model="openai/gpt-3.5-turbo", stream=False):
+        self._model = model
+        self.stream_response = stream
+
+    def __call__(self, messages, functions=[]):
+        data = {
```

Comment:
We may want to have a sensible error message that `OpenRouterAgent` requires `OPENROUTER_API_KEY` to be set for working. (JIC, This check shouldn't block using metta-motto if open router is not used)